### ISSUE TITLE
[Bug] Invalid object name 'C' when running optic diff

### ISSUE THREAD

**Issue by circleupx** [opticdev/optic#1422]
Hi,

I'm running into an odd issue with the diff command, I'm hoping someone here can provide some guidance. 

I'm following the guide [Try the optic diff command](https://www.useoptic.com/docs/#try-the-optic-diff-command), the command is successful if I compare two files in the same branch as demoed in [Two files use case](https://www.useoptic.com/docs/diff-openapi#two-files-use-case) but when I run the diff command showned below across two different git branches I get an error stating that **the object name 'C' is invalid.**

The following command.
```Powershell
optic diff todo-api.yaml --base main
```

Results in the following error.
```json
{
  stack: 'ResolverError: Command failed: git show C:Users\\circleupx\\source\\openapi-demo\\todo-api.yaml\n' +
    "fatal: invalid object name 'C'.\n" +
    ' Error opening file "Users\\circleupx\\source\\openapi-demo\\todo-api.yaml"\n' +
    '    at C:\\Users\\circleupx\\AppData\\Roaming\\npm\\node_modules\\@useoptic\\optic\\node_modules\\@useoptic\\openapi-io\\build\\parser\\resolvers\\git-branch-file-resolver.js:24:32\n' +
    '    at ChildProcess.exithandler (node:child_process:406:5)\n' +
    '    at ChildProcess.emit (node:events:520:28)\n' +
    '    at ChildProcess.emit (node:domain:475:12)\n' +
    '    at maybeClose (node:internal/child_process:1092:16)\n' +
    '    at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5)',
  code: 'ERESOLVER',
  message: 'Command failed: git show C:Users\\circleupx\\source\\openapi-demo\\todo-api.yaml\n' +
    "fatal: invalid object name 'C'.\n" +
    ' Error opening file "Users\\circleupx\\source\\openapi-demo\\todo-api.yaml"',
  source: 'Users\\circleupx\\source\\openapi-demo\\todo-api.yaml',
  path: null,
  toJSON: [Function: toJSON],
  name: 'ResolverError',
  footprint: 'null+Users\\circleupx\\source\\openapi-demo\\todo-api.yaml+ERESOLVER+Command failed: git show C:Users\\circleupx\\source\\openapi-demo\\todo-api.yaml\n' +
    "fatal: invalid object name 'C'.\n" +
    ' Error opening file "Users\\circleupx\\source\\openapi-demo\\todo-api.yaml"',
  toString: [Function: toString]
}
```

Doing some troubleshooting led me to the [git-branch-file-resolver.js ](https://github.com/opticdev/optic/blob/main/projects/openapi-io/src/parser/resolvers/git-branch-file-resolver.ts) where I can see that the code is [attempting to execute](https://github.com/opticdev/optic/blob/main/projects/openapi-io/src/parser/resolvers/git-branch-file-resolver.ts#L20) the command [git show](https://git-scm.com/docs/git-show). 

Based on my knowledge of the command git show, if you want to run git show on the same file across two different branches then you need to include the branch name in the git show command.

For example, currently, the optic CLI is attempting to execute the following command when I execute optic diff todo-api.yaml --base main

```Powershell
 git show C:Users\\circleupx\\source\\openapi-demo\\todo-api.yaml'
```

The command above should really be as follows.

```Powershell
 git show main:C\\Users\\circleupx\\source\\openapi-demo\\todo-api.yaml'
```

If update the git-branch-file-resolver.js file locally then I no longer the error "the object name 'C' is invalid".

I'm running Optic version 0.35.4.

Thank you.

**Comment by acunniffe** [comment]
Hey @circleupx you've found the problem. We definitely are not passing the right inputs to git show in your environment. This appears to be an issue specific to windows and certain shell environments. 

Can you share your shell / versions so someone can test it properly and set a fix up? We'll get this fixed up soon.

**Comment by circleupx** [comment]
Hi, 

Yes, I tried in WIndow's bash and on Powershell, and both shells gave me the same error.

Here is some information on my build and shell.

Here is the output of the command **systeminfo | findstr /B /C:"OS Name" /C:"OS Version"**

```Powershell
OS Name:  Microsoft Windows 11 Home
OS Version:   10.0.22621 N/A Build 22621
```

As previously stated, I'm on Optic 0.35.4.

For bash, here is the output of bash --version 

```bash
GNU bash, version 4.4.23(1)-release (x86_64-pc-msys)
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

For Powershell here is the output of $PSVersionTable
```Powershell
Name                           Value
----                           -----
PSVersion                      5.1.22621.608
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.22621.608
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

Let me know if you need additional information

**Comment by acunniffe** [comment]
Thanks. In the US people are coming back from thanksgiving. We'll take a look early this week and get a fix up. 

Glad WSL is unblocking you

**Comment by acunniffe** [comment]
Hey @circleupx  -- please this pre-release a try `npm install -g @useoptic/optic@0.35.6-1` 

In case you were curious about the problem, here's a quick explanation 
- git does not use filepaths -- the names you see when you make commits look like they are posix file paths, but they're really just names
- optic does not want you to have to know the name of the file in your git repo & require you to always give the full name (relative to repo root) so we were mapping the file path from the terminal input to a filename 
- the way we were doing that allowed windows paths to leak into the git names we generated for show. 

this prerelease uses `upath` https://www.npmjs.com/package/upath under the hood to do that final conversion from file name relative to repo root -> posix-like file names git uses. We're going to do a bit more testing but please let us know how this works for you.

**Comment by circleupx** [comment]
I won't be able to test until this upcoming Friday. Thanks for the explanation it helps me understand  Optic better

**Comment by circleupx** [comment]
OK, I was able to test today. I installed 0.35.6-1 as instructed, still getting the same error on both bash and Powershell.

**Comment by acunniffe** [comment]
> OK, I was able to test today. I installed 0.35.6-1 as instructed, still getting the same error on both bash and Powershell.

Hmm interesting. I'll have someone spin up the exact versions you shared earlier last week and we'll sort it out. Thanks for some of your time today trying it

**Comment by acunniffe** [comment]
We were able to reproduce on a machine we ordered and are working on a fix

**Comment by acunniffe** [comment]
There's an other pre-release out now. I've tested it in powershell several times and appears to be working. Turns out we used git show in one other place that required a fix as well. `npm install -g @useoptic/optic@0.35.7-3`

**Comment by circleupx** [comment]
> There's an other pre-release out now. I've tested it in powershell several times and appears to be working. Turns out we used git show in one other place that required a fix as well. `npm install -g @useoptic/optic@0.35.7-3`

Yep, the new fix solves that problem, however, it caused a new problem. When I execute **optic diff todo-api.yaml --base main**

I now get the following error.

![image](https://user-images.githubusercontent.com/8635797/206266497-d80b3738-ad6d-4c1d-9b6d-9017446ef5df.png)

It seem like the wrong git command is being executed. According to the error, the command being executed is **git show main::todo-api.yaml.**

The correct command is **git show main:todo-api.yaml**. There is an extra **:** in the command being executed in optic@0.35.7-3

**Comment by acunniffe** [comment]
I see it -- should be fixed in `npm install -g @useoptic/optic@0.35.7-4`. It was splicing an extra `:` into the string. Wrote a few more tests and seems to be good

**Comment by circleupx** [comment]
> I see it -- should be fixed in `npm install -g @useoptic/optic@0.35.7-4`. It was splicing an extra `:` into the string. Wrote a few more tests and seems to be good

Yep, that did it. I installed optic@0.35.7-4. Can confirm the command optic diff todo-api.yaml --base main works in bash and Powershell.

**Comment by acunniffe** [comment]
That's awesome. Thanks for all your help. We'll work this into 0.36 shortly and close this when it's released. 

Anything else I can help with?

**Comment by circleupx** [comment]
@acunniffe No, that is all, thank you so much for all the help.

**Comment by acunniffe** [comment]
Released in the latest in 0.35.7 so closing this now
